### PR TITLE
improve correction guidelines appearance and revive previews

### PR DESCRIPTION
**Pull request by huber-florian**
This pull request improves the correction guidelines appearance. Furthermore, it revives the XCode previews and the XCode does not further crash so often.

### CODE REVIEW

File: Themis/Models/Submission.swift [hunk 1]
```diff
@@ -58,8 +58,9 @@ struct SubmissionForAssessment: Codable {
     // let results: [SubmissionResult]
 }
 
-struct GradingCriteria: Codable, Identifiable {
+struct GradingCriterion: Codable, Identifiable {
     var id: Int
+    var title: String
```
Comment (pal03377) [inline Themis/Models/Submission.swift]: This crashes the app for me. Apparently, I don't get a title field. I suppose this should be optional.

Comment (pal03377) [reply Themis/Models/Submission.swift]: For reference, here's what I get when starting a new assessment:
```keyNotFound(CodingKeys(stringValue: "title", intValue: nil), Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "participation", intValue: nil), CodingKeys(stringValue: "exercise", intValue: nil), CodingKeys(stringValue: "gradingCriteria", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0)], debugDescription: "No value associated with key CodingKeys(stringValue: \"title\", intValue: nil) (\"title\").", underlyingError: nil))```

**Review by pal03377 — CHANGES_REQUESTED** [review]

File: Themis/Models/Submission.swift (hunk 1 above)
Comment (huber-florian) [reply Themis/Models/Submission.swift]: Oh, thank you, I will further explorate this.

Comment (huber-florian) [reply Themis/Models/Submission.swift]: Can you check it again? I hope it should work now.

Comment (huber-florian) [reply Themis/Models/Submission.swift]: Bildschirm­foto 2022-12-01 um 16 23 17

Comment (huber-florian) [reply Themis/Models/Submission.swift]: Bildschirm­foto 2022-12-01 um 16 23 59

Comment (huber-florian) [reply Themis/Models/Submission.swift]: Now it works with and without a title.

File: Themis/Views/Assessment/CorrectionSidebar/CorrectionGuidelinesCellView.swift [hunk 2]
```diff
@@ -10,7 +10,7 @@ import SwiftUI
 struct CorrectionGuidelinesCellView: View {
     @EnvironmentObject var assessment: AssessmentViewModel
 
-    var gradingCriteria: [GradingCriteria] {
+    var gradingCriteria: [GradingCriterion] {
         guard let criteria = assessment.submission?.participation.exercise.gradingCriteria else {
             return []
         }
```
Comment (AlexExcellence) [inline Themis/Views/Assessment/CorrectionSidebar/CorrectionGuidelinesCellView.swift:16]: Would rather put this computed property in AssessmentViewModel

File: Themis/Views/Assessment/CorrectionSidebar/CorrectionGuidelinesCellView.swift [hunk 3]
```diff
@@ -32,14 +32,11 @@ struct CorrectionGuidelinesCellView: View {
                 }.padding()
 
                 VStack(alignment: .leading, spacing: 10) {
```
Comment (AlexExcellence) [inline Themis/Views/Assessment/CorrectionSidebar/CorrectionGuidelinesCellView.swift]: Use LazyVStack instead

File: Themis/Views/Assessment/CorrectionSidebar/GradingCriteriaCellView.swift [hunk 4]
```diff
@@ -5,26 +5,60 @@
...
+                            .font(.title3)
+                    }
+
+                    HStack {
+                        Text(instruction.instructionDescription)
+                            .font(.title3)
+
+                        Spacer()
+                    }
                 }
```
Comment (AlexExcellence) [inline Themis/Views/Assessment/CorrectionSidebar/GradingCriteriaCellView.swift]: The second HStack with Spacer is unnecessary.
```suggestion
                VStack(alignment: .leading) {
                    HStack {
                        Text(instruction.gradingScale)
                            .font(.title3)
                        Spacer()
                        Text(String(format: "%.1f", instruction.credits))
                            .font(.title3)
                    }
                        Text(instruction.instructionDescription)
                            .font(.title3)
                }
```

File: Themis/Views/Assessment/CorrectionSidebar/GradingCriteriaCellView.swift [hunk 5]
```diff
@@ -5,26 +5,60 @@
...
+    func colorize(credits: Double) -> Color {
+        if credits > 0 {
+            return .green
+        } else if credits == 0.0 {
+            return .yellow
+        } else {
+            return .red
+        }
+    }
+}
```
Comment (AlexExcellence) [inline Themis/Views/Assessment/CorrectionSidebar/GradingCriteriaCellView.swift]: Put in ViewModel or  make it private

**Review by AlexExcellence — APPROVED** [review]
Everything works and looks good to me. Suggested some minor changes.

**Review by pal03377 — APPROVED** [review]
Works now 👍

**Pull request merged**